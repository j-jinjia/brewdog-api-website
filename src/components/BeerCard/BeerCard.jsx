import { React, useState } from "react";
import "./BeerCard.scss";
import ButtonViewMore from "../ButtonViewMore/ButtonViewMore";

/** This component handles the rendering of a single card. It gets passed down props from AllBeers component. On clicking View More button, will show extra information about the beer and on clicking the Return button, will go back to tis original state.*/
const BeerCard = (props) => {
  const {
    name,
    tag,
    brewYear,
    image,
    abv,
    ph,
    description,
    ingredients,
    food,
    tip,
  } = props;
  const [viewMore, setViewMore] = useState(true);
  const handleClick = () => {
    setViewMore(!viewMore);
  };
  return (
    <article className="card">
      {viewMore && (
        <>
          <img className="card__img" src={image} alt={name} />
          <div className="content">
            <h3 className="content__name">{name}</h3>
            <p className="content__tag">{tag}</p>
            <p className="content__brew-year">
              Brew Year: <br /> {brewYear}
            </p>
            <p className="content__abv">
              ABV: <br /> {abv}%
            </p>
            <p className="content__ph">
              PH:
              <br /> {ph}
            </p>
            <ButtonViewMore name="View More" handleViewMore={handleClick} />
          </div>
        </>
      )}
      {!viewMore && (
        <>
          <article className="extra-content">
            <ul className="content-list">
              <li>
                <h3 className="content-list__title">Product Description</h3>
                <p>{description}</p>
              </li>
              <li>
                <h3 className="content-list__title">Ingredients</h3>
                <p>The ingredients are : {ingredients.join(", ")}</p>
              </li>
              <li>
                <h3 className="content-list__title">Food Pairing</h3>
                <p>{food.join(" or ")}</p>
              </li>
              <li>
                <h3 className="content-list__title">Tips</h3>
                <p>{tip}</p>
              </li>
            </ul>
            <ButtonViewMore name="Return" handleViewMore={handleClick} />
          </article>
        </>
      )}
    </article>
  );
};

export default BeerCard;
