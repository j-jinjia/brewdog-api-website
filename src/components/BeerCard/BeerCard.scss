.card {
  background: linear-gradient(#aee1e1, #d3e0dc);
  margin: 1rem;
  padding: 1rem;
  width: 290px;
  height: 290px;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  border-radius: 20px;
  box-shadow: 10px 10px 5px #aaaaaa;
  overflow: auto;
  &__img {
    width: 50px;
    height: auto;
  }
  .content {
    display: grid;
    grid-template-columns: repeat(2, 1fr);
    justify-items: center;
    row-gap: 1.5rem;
    width: 60vw;
    .content__name {
      grid-column: 2 span;
      text-align: center;
    }
    .content__tag {
      grid-column: 2 span;
      text-align: center;
    }
  }
  .extra-content {
    display: flex;
    flex-direction: column;
    margin: auto;
    height: fit-content;
    width: fit-content;
    gap: 1rem;
    padding: 1rem;
    .content-list {
      list-style: none;
      gap: 1rem;
    }
  }
}
