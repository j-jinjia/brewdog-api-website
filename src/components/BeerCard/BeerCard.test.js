import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import BeerCard from "./BeerCard";

it("should match beer card component snapshot", ()=>{
    const {container} = render (<BeerCard/>)
    expect(container).toMatchSnapshot();

})

it("should render the beer card", ()=>{
   render(<BeerCard/>)
   const beerCard = screen.getByRole("article")
   expect(beerCard).toBeInTheDocument();

})
it("button should be display", ()=>{
   render(<BeerCard/>)
   const button = screen.getByRole("button")

   expect(button).toBeInTheDocument();

})
it("should render the image of the beer", ()=>{
   render(<BeerCard/>)
   const beer = screen.getByRole("img")

   expect(beer).toBeInTheDocument();

})

