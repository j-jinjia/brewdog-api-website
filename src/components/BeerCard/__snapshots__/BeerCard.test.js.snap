// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match beer card component snapshot 1`] = `
<div>
  <article
    class="card"
  >
    <img
      class="card__img"
    />
    <div
      class="content"
    >
      <h3
        class="content__name"
      />
      <p
        class="content__tag"
      />
      <p
        class="content__brew-year"
      >
        Brew Year: 
        <br />
         
      </p>
      <p
        class="content__abv"
      >
        ABV: 
        <br />
         
        %
      </p>
      <p
        class="content__ph"
      >
        PH:
        <br />
         
      </p>
      <button
        class="button"
      >
        View More
      </button>
    </div>
  </article>
</div>
`;
